src/shared/udev-util.c

```diff
@@ -704,6 +704,14 @@ int on_ac_power(void) {
 
                 bool is_battery = streq(val, "Battery");
                 if (is_battery) {
+                        r = sd_device_get_sysattr_value(d, "scope", &val);
+                        if (r < 0)
+                                log_device_debug_errno(d, r, "Failed to read 'scope' sysfs attribute, ignoring: %m");
+                        else if (streq(val, "Device")) {
```

Comment:
i wonder if the inverted check would be safer, i.e. ignore all devices except those where we see "Unknown", "System" or the field is missing. Who knows what other kind of power supplies people will add later...

Reply:
This change is "minimal" in the sense that we only filter out devices that we explicitly know are "Device". I think it's OK: I expect that if this field grows further values, it'll be something like "Hybrid" or "Multi-function", which potentially includes also powering the system.

Reply:
my thinking that maybe "port-replicator" or so might be exposed, i.e. a set of devices instead of a single one.

i guess the summary is "we don't know the unknown", hence everything is wrong.

Reply:
The kernel docs say that "Device" means "a device or a group of devices"… Dunno. We should get some kernel folks to look at this.